src/test/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandlerTest.java

```diff
@@ -504,84 +442,523 @@ public void shouldUpdateMultipleItemsWithNewStatusesIfHoldingAlreadyInPayload()
     Assert.assertNotNull(eventPayload.getContext().get(HOLDINGS.value()));
   }
 
+
   @Test
-  public void shouldNotMapItemsAndNotFillPartialErrorsIfOptimisticLockingExists()
-    throws UnsupportedEncodingException, InterruptedException, ExecutionException, TimeoutException {
-    // given
-    String firstItemId = UUID.randomUUID().toString();
-    String secondItemId = UUID.randomUUID().toString();
+  public void shouldUpdateMultipleItemsOnOLRetryAndRemoveRetryCounterFromPayloadViaSeveralRuns() throws InterruptedException, ExecutionException, TimeoutException, UnsupportedEncodingException {
 
-    JsonObject firstExistingItemJson = new JsonObject()
-      .put("id", firstItemId)
-      .put("status", new JsonObject().put("name", AVAILABLE.value()))
-      .put("materialType", new JsonObject().put("id", UUID.randomUUID().toString()))
-      .put("permanentLoanType", new JsonObject().put("id", UUID.randomUUID().toString()))
-      .put("holdingsRecordId", UUID.randomUUID().toString());
+    String itemId = UUID.randomUUID().toString();
```

Comment:
please consider refactoring to generate IDs in a loop and return them in a list

Reply:
@yaroslav-epam  good point. I will improve it.

Reply:
@yaroslav-epam fixed.